grid2op/tests/test_alert.py

```diff
@@ -13,45 +13,1234 @@
+            except Grid2OpException as e : 
+                assert e.args[0] == 'Impossible to modify the alert with your input. Please consult the documentation. The error was:\n"Grid2OpException IllegalAction "Impossible to change a raise alert id 10 because there are only 10 on the grid (and in python id starts at 0)""'
+
+
+class TestAlertBlackout(unittest.TestCase):
+    """test the basic bahavior of the assistant alert feature when a blackout occur"""
+
+    def setUp(self) -> None:
+        self.env_nm = os.path.join(
+            PATH_DATA_TEST, "l2rpn_idf_2023_with_alert"
+        )
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore")
+            kwargs_opponent = dict(lines_attacked=[ATTACKED_LINE], 
+                                   duration=10, 
+                                   steps_attack=[0,10])
+            self.env = make(self.env_nm, test=True, difficulty="1", 
+                            opponent_attack_cooldown=0, 
+                            opponent_attack_duration=99999, 
+                            opponent_budget_per_ts=1000, 
+                            opponent_init_budget=10000., 
+                            opponent_action_class=PlayableAction, 
+                            opponent_class=TestOpponent, 
+                            kwargs_opponent=kwargs_opponent)
+        self.env.seed(0)
+        self.env.reset()
+        self.do_nothing = self.env.action_space({})
+
+        self.blackout_action = self.env.action_space({})
+        self.blackout_action.gen_set_bus = [(0, -1)]
+
+    def tearDown(self) -> None:
+        self.env.close()
+# Cas avec blackout 1 ligne attaquée
+# return -10
+    def test_assistant_reward_value_blackout_attack_no_alert(self) -> None :
+        """
+        When 1 line is attacked at step 2 and we don't raise any alert
+        we expect a reward of -10 at step 3 
+        """
+        kwargs_opponent = dict(lines_attacked=[ATTACKED_LINE], 
+                                   duration=3, 
+                                   steps_attack=[2])
+        with make(self.env_nm, test=True, difficulty="1", 
+                            opponent_attack_cooldown=0, 
+                            opponent_attack_duration=99999, 
+                            opponent_budget_per_ts=1000, 
+                            opponent_init_budget=10000., 
+                            opponent_action_class=PlayableAction, 
+                            opponent_class=TestOpponent, 
+                            kwargs_opponent=kwargs_opponent, 
+                            reward_class=AlertReward(reward_end_episode_bonus=42)
+            ) as env : 
+            new_param = Parameters()
+            new_param.MAX_LINE_STATUS_CHANGED = 10
+
+            env.change_parameters(new_param)
+            env.seed(0)
+            env.reset()
+            for i in range(env.max_episode_duration()):
```

Comment:
can you remove this "for" loop and make the thing sequential (you only need to do 3 actions, 3 steps and 3 checks, it's clearer without the for loop I think)